amy/workshops/views.py

```diff
@@ -1316,13 +1321,20 @@ def form_valid(self, form):
                 request=self.request,
             )
 
+        run_instructor_training_approaching_strategy(
+            instructor_training_approaching_strategy(self.object),
+            self.request,
+            self.object,
+        )
```

Comment:
Are there cases where the first and third arguments won't reference the same object? I'm wondering if the call to `instructor_training_approaching_strategy` could be moved inside the `run_instructor_training_approaching_strategy` function body for simplicity.
Though I do see that the current way it's set up makes setting up the tests much simpler.

Reply:
The first argument is a strategy. We could calculate it e.g. above, in L1323.
The way this is set up currently we have better cohesion in both functions.